tests/storage/adapter/local_test.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+ class phpbb_storage_adapter_local_test extends phpbb_test_case
+ {
+	 protected $adapter;
+
+	public function setUp()
+ 	{
+ 		parent::setUp();
+
+		$this->adapter = new \phpbb\storage\adapter\local();
+ 	}
+
+	public function tearDown()
+	{
+		$this->adapter = null;
+	}
+
+	public function test_put_contents()
+	{
+		$this->adapter->put_contents('file.txt', '');
+		$this->assertTrue($this->adapter->exists('file.txt'));
+		$this->adapter->delete('file.txt');
+	}
+
+	public function test_get_contents()
+	{
+		$this->adapter->put_contents('file.txt', 'abc');
+		$this->assertEquals($this->adapter->get_contents('file.txt'), 'abc');
+		$this->adapter->delete('file.txt');
+	}
+
+	public function test_exists()
+	{
+		// Exists with files
+		$this->adapter->put_contents('file.txt', '');
```

Comment:
do you realize that your `put_contents()` test rely on `exists()` and that your `exists()` test rely on `put_contents()`?
To solve this issue, here you can rely on an existing file and directory (the current file and directory per example)